gateway-service/src/main/java/org/zowe/apiml/gateway/security/mapping/NativeMapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.security.mapping;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import org.zowe.commons.usermap.CertificateResponse;
+import org.zowe.commons.usermap.MapperResponse;
+import org.zowe.commons.usermap.UserMapper;
+
+/**
+ * Native on platform mapper. Depends on <a href="https://github.com/zowe/common-java/tree/v2.x.x/zos-utils">zos-utils</a> library
+ * which provides native calls to z/OS.
+ */
+@Slf4j
+@Component
+@ConditionalOnExpression("'${apiml.security.useInternalMapper:false}' == 'true'")
```

Comment:
Maybe it is more adequate to use https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnProperty.html

Reply:
ConditionalOnProperty used where applicable.